src/BBT.StrategyPattern/GenericStrategyProvider.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright © BBT Software AG. All rights reserved.
+
+namespace BBT.StrategyPattern
+{
+    using System;
+    using System.Globalization;
+    using System.Linq;
+
+    /// <summary>
+    /// See <see cref="IGenericStrategyProvider{TStrategy,TCriterion}"/>.
+    /// </summary>
+    /// <typeparam name="TStrategy">See link above.</typeparam>
+    /// <typeparam name="TCriterion">See link above.</typeparam>
+    public class GenericStrategyProvider<TStrategy, TCriterion>
+        : IGenericStrategyProvider<TStrategy, TCriterion>
+        where TStrategy : IGenericStrategy<TCriterion>
+    {
+        private readonly IStrategyLocator<TStrategy> mStrategyLocator;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GenericStrategyProvider{TStrategy, TCriterion}"/> class.
+        /// </summary>
+        /// <param name="aStrategyLocator">The <see cref="IStrategyLocator{TStrategy}"/> which can locate the strategies of which the responsible on is provided.</param>
+        public GenericStrategyProvider(IStrategyLocator<TStrategy> aStrategyLocator)
```

Comment:
Is `null` allowed for `aStrategyLocator`?  Document `null` behavior and if it is not allowed validate and throw an exception